File: src/models/RoomGroup.ts
import type { Room, RoomGroupResponse } from "src/types";
import { v4 as uuid } from "uuid";

export default class RoomGroup {
  public id: string
  public name: string
  public description: string
  public imageUrl: string
  public rooms: Room[]

  constructor(name: string) {
    this.id = uuid();
    this.name = name;
  }

  static FromRoomGroupResponse(response: RoomGroupResponse) {
    let roomGroup = new RoomGroup(response.name);
    roomGroup.id = response.id;
    roomGroup.imageUrl = response.image;
    return roomGroup;
  }
}

File: src-tauri/src/models/domain/room_group.rs
use serde::{Deserialize, Serialize};
use sqlx::{FromRow, Row};
use sqlx::sqlite::SqliteRow;
use crate::models::domain::room::Room;

#[derive(Serialize, Deserialize, Hash)]
pub struct RoomGroup {
    pub id: String,
    pub name: String,
    pub image: Option<String>,
    pub description: String,
    pub rooms: Vec<Room>
}

impl FromRow<'_, SqliteRow> for RoomGroup {
    fn from_row(row: &'_ SqliteRow) -> Result<Self, sqlx::Error> {
        Ok(Self {
            id: row.try_get("id")?,
            name: row.try_get("name")?,
            image: row.try_get("image")?,
            description: row.try_get("description")?,
            rooms: vec![]
        })
    }
}

